mergify_engine/web.py

```diff
@@ -202,157 +241,180 @@ def voluptuous_error(error):
-            data=flask.request.get_data().decode(),
-            headers={
-                "X-GitHub-Event": event_type,
-                "X-GitHub-Delivery": event_id,
-                "X-Hub-Signature": flask.request.headers.get("X-Hub-Signature"),
-                "User-Agent": flask.request.headers.get("User-Agent"),
-                "Content-Type": flask.request.headers.get("Content-Type"),
-            },
-        ).delay()
-
-    return "Event queued", 202
-
-
-@app.route("/queues/<installation_id>", methods=["GET"])
-def queues(installation_id):
-    authentification()
-
-    redis = utils.get_redis_for_cache()
+        raw = await request.body()
+        await loop.run_in_executor(
+            None,
+            forward_events.post.s(
+                config.WEBHOOK_MARKETPLACE_FORWARD_URL,
+                data=raw.decode(),
+                headers={
+                    "X-GitHub-Event": event_type,
+                    "X-GitHub-Delivery": event_id,
+                    "X-Hub-Signature": request.headers.get("X-Hub-Signature"),
+                    "User-Agent": request.headers.get("User-Agent"),
+                    "Content-Type": request.headers.get("Content-Type"),
+                },
+            ).delay,
+        )
+
+    return responses.Response("Event queued", status_code=202)
+
+
+@app.get("/queues/{installation_id}", dependencies=[fastapi.Depends(authentification)])
+async def queues(installation_id):
+    redis = await utils.get_aioredis_for_cache()
     queues = collections.defaultdict(dict)
-    filter_ = "strict-merge-queues~%s~*" % installation_id
-    for queue in redis.keys(filter_):
+    async for queue in redis.iscan(match=f"strict-merge-queues~{installation_id}~*"):
         _, _, owner, repo, branch = queue.split("~")
         queues[owner + "/" + repo][branch] = [
-            int(pull) for pull, score in redis.zscan_iter(queue)
+            int(pull) async for pull, _ in redis.izscan(queue)
         ]
 
-    return flask.jsonify(queues)
-
+    return responses.JSONResponse(status_code=200, content=queues)
 
-@app.route("/event", methods=["POST"])
-def event_handler():
-    authentification()
 
-    event_type = flask.request.headers.get("X-GitHub-Event")
-    event_id = flask.request.headers.get("X-GitHub-Delivery")
-    data = flask.request.get_json()
+@app.post("/event", dependencies=[fastapi.Depends(authentification)])
+async def event_handler(request: requests.Request):
+    event_type = request.headers.get("X-GitHub-Event")
+    event_id = request.headers.get("X-GitHub-Delivery")
+    data = await request.json()
 
-    github_events.job_filter_and_dispatch.apply_async(
-        args=[event_type, event_id, data], countdown=30
+    loop = asyncio.get_running_loop()
+    await loop.run_in_executor(
+        None,
+        functools.partial(
+            github_events.job_filter_and_dispatch.s(
+                event_type, event_id, data
+            ).apply_async,
+            countdown=30,
+        ),
```

Comment:
IIUC this is just a hack to run the function concurrently, right?

Reply:
Yes, that should be temporary as the rest will switch to async too.